src/client/dfuse/dfuse_pool.c

```diff
@@ -103,20 +106,45 @@ dfuse_pool_lookup(fuse_req_t req, struct dfuse_inode_entry *parent,
 	atomic_fetch_add(&ie->ie_ref, 1);
 	ie->ie_dfs = dfs;
 
-	rc = daos_pool_query(dfs->dffs_poh, NULL, &pool_info, NULL, NULL);
+	prop = daos_prop_alloc(0);
+	if (prop == NULL) {
+		DFUSE_LOG_ERROR("Failed to allocate pool property");
+		D_GOTO(close, rc = ENOMEM);
+	}
+
+	rc = daos_pool_query(dfs->dffs_poh, NULL, &pool_info, prop, NULL);
 	if (rc) {
 		DFUSE_TRA_ERROR(ie, "daos_pool_query() failed: (%d)", rc);
 		D_GOTO(close, rc);
 	}
 
-	ie->ie_stat.st_uid = pool_info.pi_uid;
-	ie->ie_stat.st_gid = pool_info.pi_gid;
+	/* Convert the owner information to uid/gid */
+	prop_entry = daos_prop_entry_get(prop, DAOS_PROP_PO_OWNER);
+	D_ASSERT(prop_entry != NULL);
+	rc = daos_acl_principal_to_uid(prop_entry->dpe_str,
+				       &ie->ie_stat.st_uid);
+	if (rc != 0) {
+		DFUSE_LOG_ERROR("Unable to convert owner to uid: (%d)", rc);
+		D_GOTO(close, rc);
+	}
+
+	prop_entry = daos_prop_entry_get(prop, DAOS_PROP_PO_OWNER_GROUP);
+	D_ASSERT(prop_entry != NULL);
+	rc = daos_acl_principal_to_gid(prop_entry->dpe_str,
+				       &ie->ie_stat.st_gid);
+	if (rc != 0) {
+		DFUSE_LOG_ERROR("Unable to convert owner-group to gid: (%d)",
+				rc);
+		D_GOTO(close, rc);
+	}
 
-	/* TODO: This should inspect pi_mode and corrrectly construct a correct
+	/* TODO: This should inspect ACLs and correctly construct a correct
```

Comment:
This should really only inspect owner_user owner_group and everyone.

Reply:
Yeah, that approach makes sense to me. Going to add folks from the dfuse team once CI stabilizes, so they are aware of this.